### ISSUE TITLE
Unexpected empty space is added to credentials profile name

### ISSUE THREAD

**Issue by forkercat** [aws/aws-sdk-cpp#2650]
### Describe the bug

An empty space is added to a credentials profile name when I used `AWSConfigFileProfileConfigLoader` to load and save all credentials profiles to `~/.aws/credentials`. For example, the profile name line in the file should be `[default]`, but it is `[ default]`.

This does not cause any issue in code when I used the loader to load the credentials file. However, the AWS CLI cannot recognize this format:

```
Unable to locate credentials. You can configure credentials by running "aws configure".
```

### Expected Behavior

No empty space before the profile name after saving to `~/.aws/credentials`:

```
[default]
aws_access_key_id=123
aws_secret_access_key=w123
region=us-west-2

[error_profile]
aws_access_key_id=123123
aws_secret_access_key=123123213213
region=eu-west-1
```

### Current Behavior

An empty space is added before the profile name after saving to `~/.aws/credentials`:

```
[ default]
aws_access_key_id=123
aws_secret_access_key=w123
region=us-west-2

[ error_profile]
aws_access_key_id=123123
aws_secret_access_key=123123213213
region=eu-west-1
```

### Reproduction Steps

1. Build and import AWS SDK as instructed
2. Use `AWSConfigFileProfileConfigLoader` to load and save credentials configurations
3. See the output in `~/.aws/credentials`

### Possible Solution

Did some investigation. Looks like the issue is caused by this line in [AWSConfigFileProfileConfigLoader.cpp#L567](https://github.com/aws/aws-sdk-cpp/blob/main/src/aws-cpp-sdk-core/source/config/AWSConfigFileProfileConfigLoader.cpp#L567).

```cpp
outputFile << LEFT_BRACKET << prefix << " " << profile.second.GetName() << RIGHT_BRACKET << std::endl;
```

It should be changed to:

```cpp
outputFile << LEFT_BRACKET << prefix << (m_useProfilePrefix ? " " : "") << profile.second.GetName() << RIGHT_BRACKET << std::endl;
```

`m_useProfilePrefix` is on when we use the loader to load and save non-credentials profile (e.g. `[profile default]`).

Tested locally and it worked. Putting up a PR...

### AWS CPP SDK version used

1.11.136

### Compiler and Version used

Visual Studio 2022

### Operating System and version

Windows 11

**Comment by forkercat** [comment]
PR: https://github.com/aws/aws-sdk-cpp/pull/2651

**Comment by forkercat** [comment]
^Cut a ticket to AWS CLI to fix the read issue.

**Comment by forkercat** [comment]
Checked that the changes are released in `1.11.162`

**Comment by yasminetalby** [comment]
Hello @forkercat ,

Thank you very much for your collaboration.

This issue has indeed been fixed in v[1.11.162](https://github.com/aws/aws-sdk-cpp/releases/tag/1.11.162) of the SDK. I am therefore closing this submission.

Let me know if you have any further inquiries regarding the AWS CPP SDK.

Best regards,

Yasmine
